src/providers/csharp.rs

```diff
@@ -0,0 +1,71 @@
+use super::Provider;
+use crate::nixpacks::{
+    app::App,
+    environment::{Environment, EnvironmentVariables},
+    nix::Pkg,
+    phase::{BuildPhase, InstallPhase, SetupPhase, StartPhase},
+};
+use anyhow::{Context, Result};
+
+pub struct CSharpProvider {}
+
+pub const ARTIFACT_DIR: &'static &str = &"out";
+
+impl Provider for CSharpProvider {
+    fn name(&self) -> &str {
+        "csharp"
+    }
+
+    fn detect(&self, app: &App, _env: &Environment) -> Result<bool> {
+        Ok(!app.find_files("*.csproj")?.is_empty())
+    }
+
+    fn setup(&self, _app: &App, _env: &Environment) -> Result<Option<SetupPhase>> {
+        Ok(Some(SetupPhase::new(vec![Pkg::new("dotnet-sdk")])))
+    }
+
+    fn install(&self, _app: &App, _env: &Environment) -> Result<Option<InstallPhase>> {
+        Ok(Some(InstallPhase::new("dotnet restore".to_string())))
+    }
+
+    fn build(&self, _app: &App, _env: &Environment) -> Result<Option<BuildPhase>> {
+        Ok(Some(BuildPhase::new(format!(
+            "dotnet publish --no-restore -c Release -o {ARTIFACT_DIR}"
+        ))))
```

Comment:
```suggestion
        Ok(Some(BuildPhase::new(format!(
            "dotnet publish --no-restore -c Release -o {}",
            ARTIFACT_DIR
        ))))
```

other `format!` calls in nixpacks uses this syntax